### PR TITLE
Update Text import paths: /ui

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation
Currently, there are deprecated instances of the `Text` component that use the JavaScript version of the `Box` component. This PR updates those deprecated import paths to the new component in the `ui/components/ui` folder as well as snapshots. There should be no visual or component API difference between the two. Just className changes which will touch many snapshots

* Part of #19986 
- Dependency: https://github.com/MetaMask/metamask-extension/pull/19949

## Screenshots/Screencaps
I have added Before/After screencasts to ensure no visual regressions have occurred to components with some significant updates other than snapshot updates in the code comments

## Manual Testing Steps

- Go to the latest build of storybook in this PR
- Check all components don't have visual regressions against the [hosted storybook GitHub page ](https://metamask.github.io/metamask-storybook/?path=/docs/getting-started-introduction--docs)

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/custom-spending-cap/__snapshots__/custom-spending-cap.test.js.snap [hunk 1]
```diff
@@ -24,7 +24,7 @@ exports[`CustomSpendingCap should match snapshot 1`] = `
                 class="mm-box form-field__heading-title mm-box--display-flex mm-box--align-items-baseline"
               >
                 <h6
-                  class="box mm-text mm-text--body-sm-bold box--display-inline-block box--flex-direction-row box--color-text-default"
+                  class="mm-box mm-text mm-text--body-sm-bold mm-box--display-inline-block mm-box--color-text-default"
                   for="custom-spending-cap"
                   tag="label"
                 >
```
Comment (georgewrmarshall) [inline ui/components/app/custom-spending-cap/__snapshots__/custom-spending-cap.test.js.snap:30]: Removes unused default from old Box component `box--flex-direction-row` the display of this component is inline-block. Other styles are the same

**Review by garrettbear — DISMISSED** [review]

File: ui/components/app/custom-spending-cap/__snapshots__/custom-spending-cap.test.js.snap [hunk 2]
```diff
@@ -24,7 +24,7 @@ exports[`CustomSpendingCap should match snapshot 1`] = `
                 class="mm-box form-field__heading-title mm-box--display-flex mm-box--align-items-baseline"
               >
                 <h6
-                  class="box mm-text mm-text--body-sm-bold box--display-inline-block box--flex-direction-row box--color-text-default"
+                  class="mm-box mm-text mm-text--body-sm-bold mm-box--display-inline-block mm-box--color-text-default"
                   for="custom-spending-cap"
                   tag="label"
```
Comment (georgewrmarshall) [inline ui/components/app/custom-spending-cap/__snapshots__/custom-spending-cap.test.js.snap:29]: this shouldn't be passed to label fix.

File: ui/components/ui/logo/logo.stories.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import PropTypes from 'prop-types';
 import React from 'react';
 
 import { BackgroundColor } from '../../../helpers/constants/design-system';
-import { Text } from '../../component-library/text/deprecated';
+import { Text } from '../../component-library';
 
 import Card from '../card';
 import Box from '../box';
```
Comment (georgewrmarshall) [inline ui/components/ui/logo/logo.stories.js:8]: Use Box from component-library

File: ui/components/ui/review-spending-cap/review-spending-cap.js [hunk 4]
```diff
@@ -3,8 +3,14 @@ import PropTypes from 'prop-types';
 import { I18nContext } from '../../../contexts/i18n';
 import Box from '../box';
```
Comment (georgewrmarshall) [inline ui/components/ui/review-spending-cap/review-spending-cap.js:4]: Use Box from component-library

File: ui/components/app/custom-spending-cap/custom-spending-cap.js [hunk 5]
```diff
@@ -8,8 +8,8 @@ import { addHexPrefix } from 'ethereumjs-util';
 import { I18nContext } from '../../../contexts/i18n';
 import Box from '../../ui/box';
 import FormField from '../../ui/form-field';
-import { ButtonLink, Icon, IconName } from '../../component-library';
-import { Text } from '../../component-library/text/deprecated';
+import { ButtonLink, Icon, IconName, Text } from '../../component-library';
+
 import {
   AlignItems,
   TextAlign,
```
Comment (georgewrmarshall) [inline ui/components/app/custom-spending-cap/custom-spending-cap.js:15]: No visual regressions

Screenshot 2023-07-18 at 2 53 12 PM
Screenshot 2023-07-18 at 2 53 22 PM

File: ui/components/ui/callout/callout.js [hunk 6]
```diff
@@ -4,8 +4,7 @@ import classnames from 'classnames';
 import InfoIconInverted from '../icon/info-icon-inverted.component';
 import { Severity, TextColor } from '../../../helpers/constants/design-system';
 import { MILLISECOND } from '../../../../shared/constants/time';
-import { ButtonIcon, IconName, IconSize } from '../../component-library';
-import { Text } from '../../component-library/text/deprecated';
+import { ButtonIcon, IconName, IconSize, Text } from '../../component-library';
 
 /**
  * @deprecated `<Callout />` has been deprecated in favor of the `<BannerAlert />`
```
Comment (georgewrmarshall) [inline ui/components/ui/callout/callout.js:10]: No visual regressions

Screenshot 2023-07-18 at 2 54 30 PM
Screenshot 2023-07-18 at 2 54 49 PM

File: ui/components/ui/editable-label/editable-label.js [hunk 7]
```diff
@@ -8,9 +8,13 @@ import {
...
+import {
+  ButtonIcon,
+  FormTextField,
+  IconName,
+  Text,
+  Box,
+} from '../../component-library';
 
 export default class EditableLabel extends Component {
   static propTypes = {
```
Comment (georgewrmarshall) [inline ui/components/ui/editable-label/editable-label.js:20]: No visual regressions

Screenshot 2023-07-18 at 2 58 13 PM
Screenshot 2023-07-18 at 2 58 28 PM

File: ui/components/ui/export-text-container/export-text-container.component.js [hunk 8]
```diff
@@ -22,18 +20,18 @@ function ExportTextContainer({ text = '', onClickCopy = null }) {
...
       gap={4}
     >
       <Text
-        display={DISPLAY.FLEX}
-        justifyContent={JustifyContent.CENTER}
+        display={Display.Flex}
+        justifyContent={JustifyContent.center}
         className="notranslate"
         variant={TextVariant.bodyLgMedium}
         data-testid="srp_text"
```
Comment (georgewrmarshall) [inline ui/components/ui/export-text-container/export-text-container.component.js:37]: Screenshot 2023-07-18 at 2 59 28 PM
Screenshot 2023-07-18 at 2 59 37 PM

File: ui/components/ui/form-field/form-field.js [hunk 9]
```diff
@@ -82,9 +79,8 @@ export default function FormField({
             {TitleUnitCustomComponent ||
               (titleUnit && (
                 <Text
-                  tag={TextVariant.bodySm}
-                  variant={TextVariant.bodySm}
                   as="h6"
+                  variant={TextVariant.bodySm}
                   color={TextColor.textAlternative}
                   display={Display.InlineBlock}
                 >
```
Comment (georgewrmarshall) [inline ui/components/ui/form-field/form-field.js:86]: Screenshot 2023-07-18 at 3 00 39 PM

Screenshot 2023-07-18 at 3 00 27 PM

File: ui/components/ui/icon-button/icon-button.js [hunk 10]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import { Text } from '../../component-library/text/deprecated';
+import { Text } from '../../component-library';
 import { TextVariant } from '../../../helpers/constants/design-system';
 import Tooltip from '../tooltip/tooltip';
```
Comment (georgewrmarshall) [inline ui/components/ui/icon-button/icon-button.js:7]: Screenshot 2023-07-18 at 3 12 00 PM

File: ui/components/ui/nft-info/nft-info.js [hunk 11]
```diff
@@ -40,15 +38,8 @@ export default function NftInfo({ assetName, tokenAddress, tokenId }) {
...
-            marginTop={0}
-            color={TextColor.primaryDefault}
-          >
-            {t('view')}
-          </Text>
+        <Button className="nft-info__button" variant={BUTTON_VARIANT.LINK}>
+          {t('view')}
         </Button>
       </Box>
     </Box>
```
Comment (georgewrmarshall) [inline ui/components/ui/nft-info/nft-info.js:45]: Screenshot 2023-07-18 at 3 13 32 PM
Screenshot 2023-07-18 at 3 14 02 PM

File: ui/components/ui/numeric-input/numeric-input.component.js [hunk 12]
```diff
@@ -6,7 +6,7 @@ import {
   TextVariant,
 } from '../../../helpers/constants/design-system';
 import { DECIMAL_REGEX } from '../../../../shared/constants/tokens';
-import { Text } from '../../component-library/text/deprecated';
+import { Text } from '../../component-library';
 
 export default function NumericInput({
   detailText = '',
```
Comment (georgewrmarshall) [inline ui/components/ui/numeric-input/numeric-input.component.js:12]: Screenshot 2023-07-18 at 3 15 26 PM
Screenshot 2023-07-18 at 3 15 37 PM

File: ui/components/ui/popover/popover.component.js [hunk 13]
```diff
@@ -19,8 +19,13 @@ import {
...
+import {
+  ButtonIcon,
+  Icon,
+  IconName,
+  IconSize,
+  Text,
+} from '../../component-library';
 
 const defaultHeaderProps = {
   padding: [6, 4, 4],
```
Comment (georgewrmarshall) [inline ui/components/ui/popover/popover.component.js:31]: Screenshot 2023-07-18 at 3 16 28 PM
Screenshot 2023-07-18 at 3 16 46 PM

File: ui/components/ui/radio-group/radio-group.component.js [hunk 14]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import { I18nContext } from '../../../contexts/i18n';
 import { Color, TextVariant } from '../../../helpers/constants/design-system';
-import { Text } from '../../component-library/text/deprecated';
+import { Text } from '../../component-library';
 
 function Connector({ isFirst, isLast }) {
   if (isFirst) {
```
Comment (georgewrmarshall) [inline ui/components/ui/radio-group/radio-group.component.js:9]: Screenshot 2023-07-18 at 3 17 35 PM

File: ui/components/ui/slider/slider.component.js [hunk 15]
```diff
@@ -9,7 +9,7 @@ import {
 } from '../../../helpers/constants/design-system';
 
 import InfoTooltip from '../info-tooltip/info-tooltip';
-import { Text } from '../../component-library/text/deprecated';
+import { Text } from '../../component-library';
 
 const styles = {
   root: {
```
Comment (georgewrmarshall) [inline ui/components/ui/slider/slider.component.js:15]: Screenshot 2023-07-18 at 3 18 28 PM
Screenshot 2023-07-18 at 3 18 38 PM

File: ui/pages/onboarding-flow/add-network-modal/__snapshots__/add-network-modal.test.js.snap [hunk 16]
```diff
@@ -49,8 +49,7 @@ exports[`Add Network Modal should render 1`] = `
               class="mm-box form-field__heading-title mm-box--display-flex mm-box--align-items-baseline"
             >
               <h6
-                class="box mm-text mm-text--body-sm-bold box--display-inline-block box--flex-direction-row box--color-text-default"
-                tag="label"
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/add-network-modal/__snapshots__/add-network-modal.test.js.snap:53]: removes prop being passed down to html attribute

File: ui/pages/onboarding-flow/create-password/__snapshots__/create-password.test.js.snap [hunk 17]
```diff
@@ -58,8 +58,7 @@ exports[`Onboarding Create Password Render should match snapshot 1`] = `
                 class="mm-box form-field__heading-title mm-box--display-flex mm-box--align-items-baseline"
               >
                 <h6
-                  class="box mm-text mm-text--body-sm-bold box--display-inline-block box--flex-direction-row box--color-text-default"
-                  tag="label"
+                  class="mm-box mm-text mm-text--body-sm-bold mm-box--display-inline-block mm-box--color-text-default"
```
Comment (georgewrmarshall) [inline ui/pages/onboarding-flow/create-password/__snapshots__/create-password.test.js.snap:61]: removes prop being passed down to html attribute

File: test/e2e/tests/custom-token-add-approve.spec.js [hunk 18]
```diff
@@ -159,7 +159,7 @@ describe('Create token, approve token and approve token without gas', function (
 
         const defaultSpendingCap = await driver.findElement({
           text: '7 TST',
-          css: '.box--flex-direction-row > h6',
+          css: '.mm-box > h6',
```
Comment (DDDDDanica) [inline test/e2e/tests/custom-token-add-approve.spec.js:162]: Suggestion & None blocking: can we add a test-id to this element so it won't be depending on dynamic arbitrary style-based classname?

**Review by DDDDDanica — APPROVED** [review]

File: test/e2e/tests/custom-token-add-approve.spec.js (hunk 18 above)
Comment (georgewrmarshall) [reply test/e2e/tests/custom-token-add-approve.spec.js:162]: Great call out. I admit this was a lazy fix. I've created an issue to fix after this has been merged https://github.com/MetaMask/metamask-extension/issues/20099

**Review by Mrtenz — APPROVED** [review]

**Review by garrettbear — APPROVED** [review]

**Pull request merged**